src/Extension.Tests/ChallengeTests.cs

```diff
@@ -168,17 +148,14 @@ public async Task teacher_can_access_events_when_evaluating_a_rule()
         {
             var capturedEvents = new List<List<KernelEvent>>();
             var lesson = new Lesson();
-            using var kernel = new CompositeKernel
-            {
-                new CSharpKernel()
-            }.UseLessonEvaluateMiddleware(lesson);
+            using var kernel = CreateKernel(lesson);
             var challenge = GetEmptyChallenge();
             await lesson.StartChallengeAsync(challenge);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed